src/Elcodi/Component/Cart/Entity/Interfaces/OrderInterface.php

```diff
@@ -164,7 +164,7 @@ public function getShippingMethod();
      *
      * @return $this Self object
      */
-    public function setShippingMethod($shippingMethod);
+    public function setShippingMethod(ShippingMethod $shippingMethod);
```

Comment:
This methods has an issue. $shippingMethod here is not an object, but its ID.
This means that the problem is not the casting of the variable, but the phpdoc of methods.